mods/railcraft/api/tracks/TrackRegistry.java

```diff
@@ -114,9 +113,8 @@ public void finalizeRegistry() {
         if (!RailcraftConstantsAPI.MOD_ID.equals(Loader.instance().activeModContainer().getModId())) {
             throw new TrackRegistryException("Finalize called by non-railcraft mods!");
         }
-        List<T> list = Lists.newArrayList(iterator());
-        list.sort(Comparator.comparing(T::getName));
-        if (combinations.isEmpty()) {
+        pass++; // Prevent building when not all track types or track kits are registered.
+        if (pass == 2) {
```

Comment:
pass 1: one registry finalizes;
pass 2: the other finalizes.
After both registries are freezed, we build the list.